src/wp-includes/kses.php

```diff
@@ -2320,6 +2320,8 @@ function safecss_filter_attr( $css, $deprecated = '' ) {
 			'text-indent',
 			'text-transform',
 
+			'display',
```

Comment:
I'd rather not allow `display` in it's entirety. I wasn't around for the original specification but I expect it was left out very deliberately to prevent authors and contributors from doing something like:

```html
<p style="display: none"><a href="spammy-seo-link.html">Certainly nothing dodgy here</a></p>
```

Are you able to limit it to a set of values? 

```php
'display' => array( 'flex', 'grid' )

...

// Something clever in the loop below to determine whether to use the key or value as the allowed selector if this is possible.
```

Reply:
Thanks Peter, that's exactly the kind of feedback I was hoping for on this PR! I'll have a play, yes the goal is to allow a couple of particular kinds of layouts, not to expose `display: none`, so I'll see what we can do to limit this to just what's required for the current and shorter-term goals of different layouts.

Reply:
> I wasn't around for the original specification but I expect it was left out very deliberately to prevent authors and contributors from doing something like:

Are there any discussion you could point to that would help us understand this concern?

I mean, what's stopping authors and contributors doing that by assigning the background color to text and links?

E.g.,

```html
<p class="has-white-color has-text-color has-link-color"><a href="http://test.com" data-type="URL" data-id="http://test.com">Spammy text</a></p>
</div>
```

Reply:
And if it ends up that we have to constrain values for `display`, would it be okay to have a custom check within `safecss_filter_attr` ?

Reply:
@peterwilsoncc @andrewserong How do you think we could proceed with this one? 

Would a compromise be a white list of display properties, excluding `none`, assuming the concern alluded to [above](https://github.com/WordPress/wordpress-develop/pull/2928#discussion_r912554761) is valid?

Reply:
Good question, thanks for the ping @ramonjd! For 6.1, do we need `display` to be included? I believe the only place we're currently using it is in the layout support's rendering in the Theme JSON class (here: https://github.com/WordPress/gutenberg/blob/ffd75bd84e15b7ac611d3c25bc11e0fcd4840773/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L1410)

To avoid being blocked on `display`, I was wondering if the current state of this PR is good enough to land as-is, given that it addresses the remaining layout properties that the style engine will need to validate?

Alternately, if we think it's worth it to handle `display` as well, I think adding a list of allowed values sounds like a reasonable way to go 👍

Reply:
> For 6.1, do we need display to be included?

Good question! I assumed "yes" because of the eventual migration of class-wp-theme-json-6-1.php 🤷

Reply:
>  I assumed "yes" because of the eventual migration of class-wp-theme-json-6-1.php 🤷

In the current implementation, [on this line](https://github.com/WordPress/gutenberg/blob/ffd75bd84e15b7ac611d3c25bc11e0fcd4840773/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L1435) the allow-listed display mode is output directly, rather than being parsed through `safecss_filter_attr`, so I didn't think it was needed until we move global styles output to being run through the style engine.

Reply:
Ah, thanks for clarifying! 🙇